lib/galaxy/webapps/galaxy/api/workflows.py

```diff
@@ -551,10 +556,129 @@ def build_module(self, trans, payload={}):
             'config_form'       : module.get_config_form(),
             'post_job_actions'  : module.get_post_job_actions(inputs)
         }
+        
+    @expose_api
+    def get_tool_predictions(self, trans, topk=10, max_seq_len=25, payload={}):
+        """
+        POST /api/workflows/get_tool_predictions
+        Fetch predicted tools for a workflow
+        """
+        trained_model_path = trans.app.config.model_path
+        tool_sequence = payload.get('tool_sequence', "")
+        if 'tool_sequence' not in payload or trained_model_path is None:
+            return
+        else:
+            r_file = requests.get(trained_model_path)
```

Comment:
I would prefer this doesn't use `requests.get`, and instead loads from a path on disk. As it is:

- this cannot work for users where their galaxy has no network access or is firewalled (so anyone in medical datacenters with no outgoing connections.)
- This gets re-downloaded on every request to the API, causing unnecessary load on github and potentially being seen as an abuse-type issue.

And stylistically, if you have an `if: return`, then you don't need the else, could just de-dent it.

Reply:
`requests.get` can be removed. Then, we need to keep the `trained_model` stored somewhere locally.